src/Common/parseGlobs.h

```diff
@@ -4,7 +4,9 @@
 
 namespace DB
 {
-/* Parse globs in string and make a regexp for it.
- */
-std::string makeRegexpPatternFromGlobs(const std::string & initial_str_with_globs);
+    /// Parse globs in string and make a regexp for it.
+    std::string makeRegexpPatternFromGlobs(const std::string & initial_str_with_globs);
+
+    /// Process {a,b,c...} globs separately: don't match it against regex, but generate a,b,c strings instead.
+    void expandSelector(const std::string & path, std::vector<std::string> & for_match_paths_expanded);
```

Comment:
Please return a `std::vector<std::string>` here instead of using an in-out argument.